src/coreclr/src/System.Private.CoreLib/src/System/Globalization/GlobalizationMode.cs

```diff
@@ -26,5 +32,78 @@ private static bool GetSwitchValue(string switchName, string envVariable)
 
             return ret;
         }
+
+        private static bool TryGetStringValue(string switchName, string envVariable, [NotNullWhen(true)] out string? value)
+        {
+            value = AppContext.GetData(switchName) as string;
+            if (string.IsNullOrEmpty(value))
+            {
+                value = Environment.GetEnvironmentVariable(envVariable);
+                if (string.IsNullOrEmpty(value))
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        private static void LoadAppLocalIcu(string icuSuffixAndVersion, bool suffixWithSeparator = false)
+        {
+            ReadOnlySpan<char> icuSuffix = default;
+            ReadOnlySpan<char> version = default;
+
+            // Custom built ICU can have a suffix on the name, i.e: libicuucmyapp.so.67.1
+            // So users would set the runtime switch as: myapp:67.1
+            int indexOfSeparator = icuSuffixAndVersion.IndexOf(':');
+            if (indexOfSeparator >= 0)
+            {
+                icuSuffix = icuSuffixAndVersion.AsSpan().Slice(0, indexOfSeparator);
+
+                if (icuSuffix.Length > 35)
+                {
+                    Environment.FailFast($"The resolved \"{icuSuffix.ToString()}\" suffix from System.Globalization.AppLocalIcu switch has to be < 20 chars long.");
+                }
+
+                if (icuSuffix.Length + 1 <= icuSuffixAndVersion.Length)
```

Comment:
This condition should not be needed. `Slice` will handle the empty version fine.

Reply:
but wouldn't Slice throw an exception if I go out of bounds?

Reply:
`indexOfSeparator + 1` is not going to be out of bounds for the Slice call, even when it points to the very last character.

Reply:
Right I was thinking about `":"` scenario but it would actually work just fine as it will slice from index `1` which is the length so will just return an empty span, not throw.